### PR TITLE
Few phpstan related fixes

### PR DESCRIPTION
**Pull request by Ovsyanka**
First of all i added phpstan in composer dev dependencies, because it running on travis and developers should be able to run it before pushing pull-requests.
In addition i added composer script "phpstan" and included it in "check" script.

Some ignoring errors in phpstan is actually right. And code shold be fixed.
Some appearing errors is wrong and should be ignored.
All the stuff you can see in the commits.

### CODE REVIEW

File: src/Traits/ComparisonTrait.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ public function between(ChronosInterface $dt1, ChronosInterface $dt2, $equal = t
      *
      * @param \Cake\Chronos\ChronosInterface $dt1 The instance to compare with.
      * @param \Cake\Chronos\ChronosInterface $dt2 The instance to compare with.
-     * @return static
+     * @return ChronosInterface
```
Comment (dereuromark) [inline src/Traits/ComparisonTrait.php]: Should be FQCN then.

Comment (Ovsyanka) [inline src/Traits/ComparisonTrait.php]: I am not against it, but why? Can you point me to some explanation?

Comment (cleptric) [inline src/Traits/ComparisonTrait.php]: IDE autocompletion.  Also, it's a convention we follow through the framework :)

Comment (Ovsyanka) [inline src/Traits/ComparisonTrait.php]: Okay, Fixed it

**Comment by Ovsyanka** [comment]
I see the problem with travis. I plat to exclude phpstan from dependencies if php version below 7. If you have any other suggestions - please say it!

**Comment by cleptric** [comment]
@Ovsyanka Have a look at the `.travis.yml` over at the [cakephp repo](https://github.com/cakephp/cakephp/blob/master/.travis.yml)

**Comment by dereuromark** [comment]
Can we then please also follow https://github.com/cakephp/cakephp/pull/10520 once it is merged?

**Comment by Ovsyanka** [comment]
@dereuromark, yes, I can. But I want to say my opinion about that.

I think, that it is not good to store such things it the tests folder because there expected code tests. I mean tests, that actually runs code. In our case it is phpunit tests. Phpstun is more like code style checking.

I dont think that configuration files in the root folder is bad. Because it is standard place for it. And you can put it in some different folder instead tests, like "configs", "style-checking", etc.

But it is only my opinion and you may count it or not. By now I will do as you suggested.

File: phpstan.neon [hunk 2]
```diff
@@ -1,5 +1,11 @@
 parameters:
 	ignoreErrors:
+		# Apears because interface can't describe properties and formally it doesen't have properties.
```
Comment (markstory) [inline phpstan.neon]: Typo in 'Appears'

**Comment by Ovsyanka** [comment]
Okay, It looks good now, I only have to to clear up commits history. Unfortunately, I can do it only tomorrow, but you have time to give me mooore crititism =)

**Comment by Ovsyanka** [comment]
It is done.

**Pull request merged**